test/eliminate_contiguous_test.cpp

```diff
@@ -39,4 +41,17 @@ TEST_CASE(non_standard_op)
     EXPECT(std::distance(p.begin(), p.end()) == (count - 1));
 }
 
+TEST_CASE(transpose_gemm)
+{
+    migraphx::program p;
+    auto l  = p.add_literal(get_2x2());
+    auto t  = p.add_instruction(migraphx::op::transpose{{1, 0}}, l);
+    auto c  = p.add_instruction(migraphx::op::contiguous{}, t);
+    auto ic = p.add_instruction(migraphx::op::identity{}, c);
+    p.add_instruction(migraphx::op::dot{}, ic, l);
+    auto count = std::distance(p.begin(), p.end());
+    p.compile(eliminate_contiguous_target{});
+    EXPECT(std::distance(p.begin(), p.end()) == (count - 1));
+}
```

Comment:
I see, here is the unit test.

Reply:
But I dont think this fully test the changes. In this case, contiguous can be removed, but we need a case for when contiguous cannot be removed because a later operator can't handle the different shape. There is the `pass_standard_op` test operator that might help with this. Maybe something like: transpose -> contiguous -> transpose -> transpose -> pass_standard_op.

Reply:
A few more test cases are added